### DISCUSSION TITLE
General

### DISCUSSION THREAD

**Discussion by Kai-Z-JP** [Kai-Z-JP/KaizPatchX#135] [General]
Issue建てるまでもないことを話す部屋

**Comment by Trollebas** [comment]
It's hard to say... I think everything indecent is not allowed, but everything else is.

**Comment by Saqliva** [comment]
レールの交換・重ね合わせのやつでコンフィグかなんかでShiftの押下の反転できるようにしてほしい

**Reply by Kai-Z-JP** [reply]
実装がサーバー側だからサーバー単位でのコンフィグ設定になるし厳しいかなぁ

**Comment by ringo-1234** [comment]
Kaiz様

私はRTM1.12.2の改造版の開発に取り組んでおります者でございます。
KaizPatchにおける素晴らしい車両ミニチュア化機能を、私のプロジェクトに移植させていただきたく試みておりますが、車両エンティティをミニチュア化する機能の詳細につきまして、私の技術不足により理解が及んでおりません。つきましては、実装方法をご教示いただけますと大変ありがたく存じます。

以下の点についてお伺いできれば幸いです:
・ミニチュア機能の実装において重要な情報として、どのクラスにおいてどのような処理を行うことで車両のミニチュア化を実現されているか(主要なクラスをご教示いただけますと幸いです)

また、もしこちらのDiscussionsでのご相談が不適切でございましたら、別の方法をご指定いただければ幸いです。ご返信が難しい場合は、どうぞお気になさらずスルーしていただいて構いません。

お忙しいところ誠に恐縮ではございますが、ご協力を賜れますと幸甚に存じます。何卒よろしくお願い申し上げます。
長文失礼しました。

**Reply by Kai-Z-JP** [reply]
MCTEWorldという仮想ワールドを描画しているような感じだったはずですが、1.7.10と1.12.2だと描画の仕組みがかなり違う（らしい）ので難しいと思います。
実は自分も詳しい仕組みはよくわかっていないためお力添えは難しいかもしれません。

参考までに描画クラスのリンクを置いておきます
https://github.com/Kai-Z-JP/KaizPatchX/blob/master/src/main/java/jp/ngt/mcte/block/RenderMiniature.java

**Reply by ringo-1234** [reply]
お忙しい中、ご返信いただきありがとうございます。
MCTEWorldという仕組みだったのですね。バージョン間のレンダリングの違いで難しいとのこと、承知いたしました。
わざわざお答えいただき感謝いたします。いただいた情報を参考に、自分でも研究してみます。
ありがとうございました。

**Reply by Kai-Z-JP** [reply]
具体箇所で実装の不明点等あれば、可能な限りは説明しますのでまたご連絡いただけると。
